### PR TITLE
Clean up for status bar indicator code.

### PR DESCRIPTION
**Pull request by pykong**
- Enhanced readability.
- Cleaner, more pythonic code.

### CODE REVIEW

**Comment by pykong** [comment]
Generators for the rescue!

**Comment by braver** [comment]
Eh, we must've improved performance again because I can't get the darn thing to show up anymore. Code looks fine to me.

File: busy_indicator_view.py [hunk 1]
```diff
@@ -56,23 +56,18 @@ def on_activated_async(self, active_view):
...
-    'Linting.  ',
-    'Linting.. ',
-    'Linting. .',
-    'Linting ..',
-    'Linting  .',
-]
+def tick():
+    while True:
+        for phase in ('.  ', '.. ', '. .' ' ..', '  .'):
+            yield "Linting" + phase
```
Comment (kaste) [inline busy_indicator_view.py]: In python you actually do this with `itertools.cycle`

```
phaser = cycle(['.  ', '.. ', '. .' ' ..', '  .'])
```

And then you call next(phaser)

File: busy_indicator_view.py [hunk 2]
```diff
@@ -56,23 +56,18 @@ def on_activated_async(self, active_view):
...
 
 def draw(active_view, running, **kwargs):
     buffer_id = active_view.buffer_id()
     start_time = running.get(buffer_id, None)
     now = time.time()
     if start_time and (INITIAL_DELAY <= (now - start_time) < TIMEOUT):
-        num = len(indicators)
-        text = indicators[int((now - start_time) * 1000 / CYCLE_TIME) % num]
-        active_view.set_status(STATUS_BUSY_KEY, text)
+        active_view.set_status(STATUS_BUSY_KEY, next(tick()))
```
Comment (kaste) [inline busy_indicator_view.py]: This doesn't work bc every time you draw you start a new generator `tick()` and take one value out of it via `next()`.

File: busy_indicator_view.py [hunk 3]
```diff
@@ -56,23 +56,18 @@ def on_activated_async(self, active_view):
...
         sublime.set_timeout_async(lambda: draw(**State), CYCLE_TIME)
```
Comment (kaste) [inline busy_indicator_view.py]: If you do animations, probably in JS or so, you always reference a timer. `now` `then` etc.

Bc you actually never know when you're callback runs. So you actually don't need a cycler that gives you a new frame per call. You have to write a function that gives you the correct frame at given time.

File: busy_indicator_view.py (hunk 1 above)
Comment (pykong) [reply busy_indicator_view.py]: @kaste +1 for cycle My previous indicator was implemented exactly that way ()!
And you ripped it out. ;-)

**Comment by pykong** [comment]
OK. Was a suggestion. Let's move on to other PRs.

**Pull request closed without merging**